Fortran90/data_types/funcv.f90

```diff
@@ -12,18 +7,23 @@ subroutine funcv(fvec)
 	real*8, intent(out) :: fvec(npar)
 	
         do i=1,npar
+        ! "total" accumulates the number of non-zero elements into the
+        ! row vector of "obj_approx%eqn%lhs". This is equal to
+        ! obj_approx%eqn%rhs, so we just count it once.
          total=0
          do j=1,nsites
           if(obj_approx%eqn%lhs(i,j).ne.0) then
             total=total+1
           end if
          end do
+        ! Knowing that, we allocate the vectors to calculate
+        ! correlations.
          allocate(v1(total),v2(total)) 
          do j=1,total
           v1(j)=obj_approx%eqn%lhs(i,j)
           v2(j)=obj_approx%eqn%rhs(i,j)
          end do
-         fvec(i)=log(obj_approx%corfun(v1,total,obj_approx))-log(obj_approx%corfun(v2,total,obj_approx))
+         fvec(i)=obj_approx%corfun(v1,total,obj_approx)-obj_approx%corfun(v2,total,obj_approx)
```

Comment:
Why did you remove the logarithms here? I don't get solutions for chemical potentials higher than about -0.3 and I believe variables may be overflowing. I'd say add the logs back in.